### PR TITLE
create connect user

### PR DESCRIPTION
**Pull request by calellowitz**
## Technical Summary

Builds on https://github.com/dimagi/commcare-hq/pull/33300 to allow the creation of mobile workers that are linked to connect id users via the API.

### Safety story

This does not change any existing functionality, just adds a further option.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by millerdev** [comment]
> This does not change any existing functionality, just adds a further option.

If `connect_username` is present it will do a thing that was not done before. That seems like a change to existing functionality. Do you think this feature is something we will want to keep? If so, tests should be added to ensure it works.

**Comment by calellowitz** [comment]
> > This does not change any existing functionality, just adds a further option.
> 
> If `connect_username` is present it will do a thing that was not done before. That seems like a change to existing functionality. Do you think this feature is something we will want to keep? If so, tests should be added to ensure it works.

Yeah, sorry. That was a poor choice of words. I meant that existing use cases should continue to work as before (this should have no impact on current workflows). Agree that adding a test for the new workflow is a good idea.

**Comment by calellowitz** [comment]
@snopoke This is ready for review again. I didn't realize it had slipped through the cracks

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -207,6 +208,7 @@ class Meta(v0_1.CommCareUserResource.Meta):
         detail_allowed_methods = ['get', 'put', 'delete']
         list_allowed_methods = ['get', 'post']
         always_return_data = True
+        authentication = RequirePermissionAuthentication(HqPermissions.edit_commcare_users)
```
Comment (snopoke) [inline corehq/apps/api/resources/v0_5.py:211]: Doesn't this get inherited from the parent class?

**Review by snopoke — APPROVED** [review]

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (calellowitz) [reply corehq/apps/api/resources/v0_5.py:211]: It does, I meant to delete it when I removed the custom auth

**Comment by calellowitz** [comment]
I added a feature flag check so we could control which domains can use the new arg

**Review by snopoke — APPROVED** [review]

**Pull request merged**